CMakeLists.txt

```diff
@@ -102,6 +106,64 @@ elseif(CMAKE_SYSTEM_NAME STREQUAL "Windows")
 elseif(CMAKE_SYSTEM_NAME STREQUAL "Generic")
     if(PLATFORM_NAME STREQUAL "nuttx")
         set(PLATFORM_NAME_NUTTX ON)
+        set(TRANSPORT_TYPES "UDP" "TCP" "SERIAL")
+    endif()
+endif()
+
+# Check external transport.
+SET(PLATFORM_EXTERNAL_TRANSPORT_HEADER "" CACHE PATH "External transports header file")
+SET(PLATFORM_EXTERNAL_TRANSPORT "" CACHE PATH "External transports code file")
+
+if( PLATFORM_EXTERNAL_TRANSPORT)
+    set(PLATFORM_EXTERNAL ON)
+
+    set(TRANSPORT_TYPES "SERIAL")
```

Comment:
Could we prepare it for `UDP` and `TCP` transports?

Reply:
Maybe with a `-DEXTERNAL_TYPE` flag?